### PR TITLE
Fine tune layer selector

### LINKED ISSUE #10093: Legends should hide when their layer is hidden
Right now, if I hide a layer, the legend keeps on showing. 

screen shot 2016-10-07 at 17 45 39

The expected behaviour is that whenever I hide a layer, the legends of that layer should hide as well.

### LINKED ISSUE #10109: Legend does not fade out when disabling view from layer list
### STR

* Add a legend to a layer via changing its style.
* Disable layer view. The layer selector checkbox is disabled, but the legend does not fade out (as you can check in this [embed](https://solutionscdb.carto.com/u/cdbsol-admin/builder/9c060f60-8ee0-11e6-b675-0ee66e2c9693/embed)). 
* In addition, when turning on the checkbox on the layer selector, the layer continues to be on "non-view" status. So it looks like the layer selector and the layer views are not quite well connected.

### PR DESCRIPTION
**Pull request by nobuti**
This PR fixes #10093 and fixes #10109 on builder side.

layer-selector

It depends on https://github.com/CartoDB/cartodb.js/pull/1447

CR @xavijam

### CODE REVIEW

**Comment by xavijam** [comment]
LGTM! 🇪🇸

File: lib/assets/javascripts/cartodb3/deep-insights-integrations.js [hunk 1]
```diff
@@ -87,6 +91,16 @@ var F = function (opts) {
 
   this.visMap().on('change', _.debounce(this._saveStateDefinition.bind(this), 500), this);
 
+  // In order to sync layer selector and layer visbility
+  this._getLayers().on('change:visible', function (layer, visible) {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/deep-insights-integrations.js]: Don't we need to add this bind each time a layer is added?
Unbind when remove too, of course.

Comment (nobuti) [inline lib/assets/javascripts/cartodb3/deep-insights-integrations.js]: We are binding on the collection.

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/deep-insights-integrations.js]: 🇪🇸

**Pull request merged**